ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,177 @@
+"""Collect insights and upload it to the Ingress service.
+
+It collects the data on startup and then it waits for the specified interval.
+"""
+
+# HACK for debug launch - ols import
+# import sys
+# from pathlib import Path
+# current_dir = Path(__file__).parent.absolute()
+# sys.path.insert(0, str(current_dir.parent.parent))
+
+import io
+import logging
+import os
+import pathlib
+import tarfile
+import time
+
+import requests
+
+from ols.user_data_collection import auth
+
+OLS_USER_DATA_PATH = os.environ.get("OLS_USER_DATA_PATH", "./user-data/")
+
+# TODO: move these envs to config?
+OLS_USER_DATA_COLLECTION_INTERVAL = int(
+    os.environ.get("UDC_COLLECTION_INTERVAL", 2 * 60 * 60)
+)  # 2 hours in seconds
+
+# available ingress urls are:
+# - https://console.redhat.com/ - production
+# - https://console.stage.redhat.com/ - staging/dev
+# NOTE: Token is not working for stage...
+INGRESS_URL = os.environ.get("INGRESS_URL", "https://console.redhat.com/")
+INGRESS_ENDPOINT = os.environ.get("INGRESS_ENDPOINT", "api/ingress/v1/upload")
+INGRESS_TIMEOUT = 10  # seconds
+
+# customer portal offline token is generated from https://access.redhat.com/management/api
+CP_OFFLINE_TOKEN = os.environ.get("CP_OFFLINE_TOKEN")
+
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+def collect_ols_data_from(location: str) -> list[pathlib.Path]:
+    """Collect files from a given location.
+
+    Args:
+        location: Path to the directory to be searched for files.
+
+    Returns:
+        List of paths to the collected files.
+
+    Only JSON files from the 'feedback' and 'transcripts' directories are collected.
+    """
+    files = []
+
+    files += list(pathlib.Path(location).glob("feedback/*.json"))
+    files += list(pathlib.Path(location).glob("transcripts/*/*/*.json"))
+
+    return files
```

Comment:
What's the difference between feedback and transcripts?

Reply:
feedback is the user telling this "this response was good/bad because....", along w/ the specific interaction the user had (what they asked, what the llm responded with)

transcripts are us recording all the interactions between the user and the llm

Reply:
Understood. Thank you